backend/src/routes/aggregators/applicationsArgo.ts

```diff
@@ -0,0 +1,161 @@
+  name: string
+  namespace: string
+  created: string
+  destinationNamespace: string
+  destinationName: string
+  destinationCluster: string
+  destinationServer: string
+  path: string
+  repoURL: string
+  targetRevision: string
+  chart: string
+  cluster: string
+}
+
+export async function getArgoApps() {
+  const argoAppSet = new Set<string>()
+  const localApps = getLocalArgoApps(argoAppSet)
+  const remoteApps = await getRemoteArgoApps(argoAppSet)
+  return { argoApps: localApps.concat(remoteApps), argoAppSet }
+}
+
+function getLocalArgoApps(argoAppSet: Set<string>) {
+  const argoApps = getKubeResources('Application', 'argoproj.io/v1alpha1')
+  return argoApps.filter((app) => {
+    const argoApp = app as IArgoAppLocalResource
+    const resources = argoApp.status ? argoApp.status.resources : undefined
+    const definedNamespace = resources?.[0].namespace
+
+    // cache Argo app signature for filtering OCP apps later
+    argoAppSet.add(
+      `${argoApp.metadata.name}-${
+        definedNamespace ? definedNamespace : argoApp.spec.destination.namespace
+      }-local-cluster)}`
+      // }-${getArgoDestinationCluster(argoApp.spec.destination, managedClusters, 'local-cluster')}` //TODO
+    )
+    const isChildOfAppset =
+      argoApp.metadata.ownerReferences && argoApp.metadata?.ownerReferences[0].kind === 'ApplicationSet'
+    if (!argoApp.metadata.ownerReferences || !isChildOfAppset) {
+      return true
+    }
+    return false
+  })
+}
+
+async function getRemoteArgoApps(argoAppSet: Set<string>) {
+  const options = await getServiceAccountOptions()
+  const argoApps = await getPagedRemoteArgoApps(options)
+
+  const apps: IResource[] = []
+  argoApps.forEach((argoApp: IArgoAppRemoteResource) => {
+    argoAppSet.add(`${argoApp.name}-${argoApp.destinationNamespace}-${argoApp.cluster}`)
+    if (!argoApp._hostingResource) {
+      // Skip apps created by Argo pull model
+      apps.push({
+        apiVersion: 'argoproj.io/v1alpha1',
+        kind: 'Application',
+        metadata: {
+          name: argoApp.name,
+          namespace: argoApp.namespace,
+          creationTimestamp: argoApp.created,
+        },
+        spec: {
+          destination: {
+            namespace: argoApp.destinationNamespace,
+            name: argoApp.destinationName,
+            server: argoApp.destinationCluster || argoApp.destinationServer,
+          },
+          source: {
+            path: argoApp.path,
+            repoURL: argoApp.repoURL,
+            targetRevision: argoApp.targetRevision,
+            chart: argoApp.chart,
+          },
+        },
+        status: {
+          cluster: argoApp.cluster,
+        },
+      } as IResource)
+    }
+  })
+
+  return apps
+}
+
+// get argo apps from search api in three queries with a second apart
+async function getPagedRemoteArgoApps(options: RequestOptions) {
+  // a,i,n,etc are used with the hope that the distribution of names is evenly matched between queries
```

Comment:
Most resource names permit a numeric character as the first character, including Application for Argo CD, so this scheme to ease the load on search will miss some applications  in its current form (See https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names; by attempting to create an Application with an uppercase character, I gleaned from the error message that they use RFC 1123)

I wonder if this is necessary? Perhaps we should leave it out for now or work with search to provide a real paging solution.